admin-ui/app/routes/index.tsx

```diff
@@ -1,33 +1,38 @@
-import { useState, useEffect, lazy, Suspense } from 'react'
+import { useState, useEffect } from 'react'
 import { Route, Routes, Navigate } from 'react-router-dom'
 import { useSelector } from 'react-redux'
 
-import NavbarOnly from './Layouts/NavbarOnly'
-import SidebarDefault from './Layouts/SidebarDefault'
-import SidebarA from './Layouts/SidebarA'
-import SidebarWithNavbar from './Layouts/SidebarWithNavbar'
-
 // ----------- Layout Imports ---------------
-import { processRoutes } from 'Plugins/PluginMenuResolver'
+import { processRoutes, processRoutesSync } from 'Plugins/PluginMenuResolver'
 
 import GluuSuspenseLoader from 'Routes/Apps/Gluu/GluuSuspenseLoader'
 
 import { uuidv4 } from 'Utils/Util'
 import ProtectedRoute from './Pages/ProtectRoutes'
-
-const DashboardPage = lazy(() => import('./Dashboards/DashboardPage'))
-const ProfilePage = lazy(() => import('./Apps/Profile/ProfilePage'))
-const Gluu404Error = lazy(() => import('./Apps/Gluu/Gluu404Error'))
-const ByeBye = lazy(() => import('./Pages/ByeBye'))
-const GluuNavBar = lazy(() => import('./Apps/Gluu/GluuNavBar'))
-const DefaultSidebar = lazy(() => import('./../layout/components/DefaultSidebar'))
+import { LazyRoutes } from 'Utils/RouteLoader'
 
 //------ Route Definitions --------
 
 export const RoutedContent = () => {
   const [pluginMenus, setPluginMenus] = useState<Array<any>>([])
+  const [loading, setLoading] = useState(true)
+
   useEffect(() => {
-    setPluginMenus(processRoutes())
+    const loadPlugins = async () => {
+      try {
+        setLoading(true)
+        const routes = await processRoutes()
+        setPluginMenus(routes)
+      } catch (error) {
+        console.error('Failed to load plugins:', error)
+        // Fallback to sync loading
+        setPluginMenus(processRoutesSync())
+      } finally {
+        setLoading(false)
+      }
+    }
+
+    loadPlugins()
   }, [])
```

Comment:
Remove unused loading state.
loading is set/cleared but never used in render. Drop the state and setLoading calls to reduce noise.

Reply:
done